datadog.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// Package datadog contains a Datadog exporter.
+package datadog
+
+import (
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	defaultHost   = "localhost"
+	defaultPort   = "8125"
+	opencensusTag = "source:Opencensus"
+)
+
+var (
+	exporter *Exporter
+	tags     = []string{opencensusTag}
+	reg      = regexp.MustCompile("[^a-zA-Z0-9]+")
+)
+
+// Exporter exports stats to Datadog.
+type Exporter struct {
+	opts      Options
+	collector *collector
+}
+
+// ExportView exports to Datadog if view data has one or more rows.
+func (e *Exporter) ExportView(vd *view.Data) {
+	if len(vd.Rows) == 0 {
+		return
+	}
+	e.collector.addViewData(vd)
+}
+
+// Options contains options for configuring the exporter.
+type Options struct {
+	// Namespace specifies the namespace to which metrics are appended.
+	Namespace string
+
+	// Host for DogStatsD connection
+	Host string
+
+	// Port for DogStatsD connection
+	Port string
+
+	// OnError will be called in the case of an error while uploading the stats.
+	// If not set, errors are simply logged.
+	OnError func(err error)
+
+	// Tags specifies a set of global tags to attach to each metric.
+	Tags []string
+}
+
+// getEndpoint determines the address for the statsD client
+func (o *Options) getEndpoint() string {
+	host, port := o.Host, o.Port
+	if host == "" || port == "" {
+		host, port = defaultHost, defaultPort
+	}
+	endpoint := host + ":" + port
+	return endpoint
+}
+
+func (o *Options) onError(err error) {
+	if o.OnError != nil {
+		o.OnError(err)
+	} else {
+		log.Printf("Failed to export to Datadog: %v\n", err)
+	}
+}
+
+// NewExporter returns an exporter that exports stats to Datadog
+func NewExporter(o Options) *Exporter {
+	endpoint := o.getEndpoint()
+
+	client, err := statsd.New(endpoint)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	collector := &collector{
```

Comment:
I would put this into a function called `newCollector` (or `newStatsExporter`), such as:
```go
func newStatsExporter(o Options) *statsExporter {
    client, err := statsd.New(o.StatsAddr)
    if err != nil {
        log.Fatal(err)
    }
    return &collector{
        opts: o,
        viewData: make(map[string]*view.Data),
        client: client,
    }
}
```
Then, `NewExporter` will become:
```go
func NewExporter(opts Options) *Exporter {
    return &Exporter{statsExporter: newStatsExporter(opts)}
}
```
This will make it easier to extend it once #3 gets merged and we add the `traceExporter`